spec/models/miq_widget_spec.rb

```diff
@@ -184,7 +184,7 @@
         end
 
         it 'only returns groups in the current region' do
-          groups = [@group1, @group2]
+          groups = MiqGroup.where(:id => [@group1, @group2].map(&:id))
```

Comment:
I don't follow how this fixes what you've described,  Since you are doing `.map(&:id)` you're still returning an Array and not a Relation.  If you all you need is an Array of ids, can you do `groups = [@group1.id, @group2.id]`

Reply:
The key is the wrapping:

`MiqGroup.where(:id => [1,2]) # Returns an activerecord relation`

as opposed to:

`[group1, group2].where # invalid`

Although, experimentation shows that I don't need to map the id's, I can just do this:

`MiqGroup.where(:id => [group1, group2])`